rover_gui/src/QTopUtilityBar/QTopUtilityBar.cpp

```diff
@@ -0,0 +1,365 @@
+        }
+    }
+
+    emit this->updateTimer(secondsBeforeTimeout);
+}
+
+void QTopUtilityBar::onUpdateBatteryUI(float pourcent_)
+{
+    _ui.batteryLabel->setText(QString::number(static_cast<int>(pourcent_)) + " %");
+    QIcon icon;
+
+    if (pourcent_ >= 85)
+    {
+        icon = QIcon(":/icons/BatteryIcon100.svg");
+    }
+    else if (pourcent_ >= 55)
+    {
+        icon = QIcon(":/icons/BatteryIcon75.svg");
+    }
+    else if (pourcent_ >= 40)
+    {
+        icon = QIcon(":/icons/BatteryIcon50.svg");
+    }
+    else if (pourcent_ >= 20)
+    {
+        icon = QIcon(":/icons/BatteryIcon25.svg");
+    }
+    else
+    {
+        icon = QIcon(":/icons/BatteryIcon0.svg");
+    }
+
+    _ui.batteryIcon->setIcon(icon);
+}
+
+void QTopUtilityBar::onUpdateWifiUI(float rssi_, float speed_)
+{
+    _ui.signalQualityLabel->setText("RSSI: " + QString::number(static_cast<int>(rssi_)));
+    _ui.connectionSpeedLabel->setText(QString::number(static_cast<float>(speed_), 'f', 1) + " Mb/s");
+
+    QIcon icon;
+
+    if (rssi_ <= -85)
+    {
+        icon = QIcon(":/icons/RSSI_one.png");
+    }
+    else if (rssi_ > -85 && rssi_ <= -75)
+    {
+        icon = QIcon(":/icons/RSSI_two.png");
+    }
+    else if (rssi_ > -75 && rssi_ <= -65)
+    {
+        icon = QIcon(":/icons/RSSI_three.png");
+    }
+    else
+    {
+        icon = QIcon(":/icons/RSSI_four.png");
+    }
+
+    _ui.RSSILabel->setIcon(icon);
+}
+
+void QTopUtilityBar::onUpdateGNSS(uint8_t fix_, float heading_, uint8_t satNbr_, float long_, float lat_)
+{
+    _ui.HeadingLabel->setText(QString::number(static_cast<float>(heading_), 'f', 2) + " deg   ");
+    _ui.satellitesNbrLabel->setText(QString::number(static_cast<int>(satNbr_)) + "   ");
+    _ui.latitudeLabel->setText("Lat: " + QString::number(static_cast<float>(lat_), 'f', 6));
+    _ui.longitudeLabel->setText("Long: " + QString::number(static_cast<float>(long_), 'f', 6));
+
+    _ui.satellliteIcon_pb->setIcon(QIcon(":/icons/GNSSIcon.svg"));
+    _ui.headingIcon_pb->setIcon(QIcon(":/icons/HeadingIcon.svg"));
+
+    std::string fixQuality;
+
+    switch (fix_)
+    {
+        case 0:
+            fixQuality = "NF";
+            break;
+        case 1:
+            fixQuality = "GPS";
+            break;
+        case 2:
+            fixQuality = "DGPS";
+            break;
+        case 3:
+            fixQuality = "PPS";
+            break;
+        case 4:
+            fixQuality = "RTK";
+            break;
+        case 5:
+            fixQuality = "RTK-F";
+            break;
+        case 6:
+            fixQuality = "EST";
+            break;
+        case 7:
+            fixQuality = "MAN";
+            break;
+        case 8:
+            fixQuality = "SIM";
+            break;
+        default:
+            fixQuality = "UNK";
+            break;
+    }
+
+    _ui.GNSSFixLabel->setText("Fix Quality: " + QString::fromStdString(fixQuality));
+}
+
+void QTopUtilityBar::onUpdateTimer(int secondsBeforeTimeOut_)
```

Comment:
Avoid using "magic numbers here for timeouts. Explicit static constexp 
```suggestion
static constexpr float UPDATE_TIMER_TIMEOUT = 3000.0F;
```

Reply:
dnoe